nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/script/impl/JythonScriptEngineConfigurator.java

```diff
@@ -50,7 +50,7 @@ public String getScriptEngineName() {
     @Override
     public Object init(ScriptEngine engine, String scriptBody, String[] modulePaths) throws ScriptException {
         // Always compile when first run
-        if (engine != null && compiledScriptRef.get() == null) {
```

Comment:
So this change would address the issue at hand, I believe, but it would do so at the expense of requiring that ExecuteScript would have to recompile the Jython script for every invocation of `onTrigger`. That could result in some extremely poor performance. I think we should instead leave this check in, but add an onPropertyModified method for the script components that is responsible for resetting the state here. So that way, we only need to recompile if a property is changed.